Sources/AWSSDKSwiftCore/ServiceConfig.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  ServiceConfig.swift
+//  AWSSDKSwiftCore
+//
+//  Created by Fabian Fett on 11.02.20.
+//
+
+import class Foundation.ProcessInfo
+
+public struct ServiceConfig {
+    public let region            : Region
+    public let amzTarget         : String?
+    public let service           : String
+    public let signingName       : String
+    public let serviceProtocol   : ServiceProtocol
+    public let apiVersion        : String
+    public let endpoint          : String
+    public let serviceEndpoints  : [String: String]
+    public let partitionEndpoint : String?
+    public let possibleErrorTypes: [AWSErrorType.Type]
+    
```

Comment:
You'll need to add middlewares here as well. Services can have specific middleware. We'll probably need middleware arrays on both client and service.